var/spack/repos/builtin/packages/py-cdat-lite/package.py

```diff
@@ -40,3 +40,14 @@ class PyCdatLite(PythonPackage):
     depends_on("python@2.5:2.8", type=('build', 'run'))
     depends_on("py-numpy", type=('build', 'run'))
     depends_on('py-setuptools', type='build')
+
+    phases = ['install']
+
+    def install(self, spec, prefix):
+        """Install everything from build directory."""
+        install_args = self.install_args(spec, prefix)
+        # Combine all phases into a single setup.py command,
+        # otherwise extensions are rebuilt without rpath by install phase:
```

Comment:
Hi @adamjstewart , I think the need to combine phases is limited to this package. I didn't have the same trouble with py-pillow.

I think the `--rpath` option is part of py-setuptools, but it is only available for the `build_ext` phase. So to use it generally, we would have to include the build_ext phase by default.

Perhaps the more general solution would be to rely on the spack compiler wrappers to set rpath. But the wrappers are deliberately replaced by references to the real compilers during the python installation procedure. The reason seems to be to allow users to install their own packages without spack.

I am wondering if we need a way to run actual compiler versions outside of spack while still making use of the spack wrappers to define rpath correctly. For example, when a package like python is installed, the associated compiler could be represented by a wrapper script inside the python package. Then when python extensions are installed, the python compiler would invoke the correct compiler via the spack compiler wrappers, so that rpath and other search paths would always be set correctly. Does that idea make any sense?